### ISSUE TITLE
threshold has negligible effect

### ISSUE THREAD

**Issue by eafpres** [plemeri/transparent-background#43]
System:
WSL 2 in Windows 11, running Ubuntu 20.04
Python 3.9
3080 Ti 12 GB GPU
32 GB RAM

Excellent package.  This is the best OSS background removal I have found, thanks.

I have tested using threshold 0.01 and 0.99 to see if I can optimize for my use case.
Two pairs of images as example--left is original, right after remover.process()
A) Threshold 0..01
![image](https://github.com/plemeri/transparent-background/assets/30508983/393ed1c7-1da4-4286-b3cd-ecdbdcff9d1e)

B) Threshold 0.99
![image](https://github.com/plemeri/transparent-background/assets/30508983/7d45c36c-e130-435b-b5c0-2645c4aeb878)

Here we see that we are having trouble with light background pavement, and over the full range of threshold cannot meaningfully impact that performance.

Is there anything else I could try with the code as is, or should I consider some form of fine-tuning?  Do you support that?

**Comment by plemeri** [comment]
Hi @eafpres, thank you for choosing our work. Glad to be your help!

First of all, I tried myself with the provided image and here are the results.
Threshold|0.5 (default)|0.01|0.99
:-:|:-:|:-:|:-:
![test2](https://github.com/plemeri/transparent-background/assets/40786892/d397c2ef-cf73-4c95-88ca-88a6af95a2d9) | ![test2_rgba_threshold_050](https://github.com/plemeri/transparent-background/assets/40786892/df500235-455d-44a5-a9e4-c988d0b9e289) | ![test2_rgba_threshold_001](https://github.com/plemeri/transparent-background/assets/40786892/2fc7ed05-492a-4d93-a9af-54bb6bb1c230) | ![test2_rgba](https://github.com/plemeri/transparent-background/assets/40786892/4e69dad0-8272-471c-9400-c3e3f6c728c2)
Saliency Map Only |  ![test2_map_theshold_050](https://github.com/plemeri/transparent-background/assets/40786892/972f6153-64cc-4994-bb91-fc9283b1219e) | ![test2_map_threshold_001](https://github.com/plemeri/transparent-background/assets/40786892/22df3187-f70f-4cff-9784-23ba0cc2e3c8) | ![test2_map_threshold_099](https://github.com/plemeri/transparent-background/assets/40786892/353b7e87-0065-4e89-9869-6f42644f57c2)

There are different from your results. Can I have your script since I used a command line tool not python script. It seems like you used our python API, so it would be helpful for us to find a problem if you provide your script.

Moreover, we also recommend trying `base-nightly` mode which can be enabled by using `--mode base-nightly` argument.

Feel free to ask more question. Thanks.

**Comment by eafpres** [comment]
> There are different from your results

The original image is 3000 x 4000 pixels .jpg
I wonder if the large image has some relation to the way the threshold is working?
I did do further testing and I can see impact if I make the threshold 1e-4 or 1e-3 in these cases.

> It seems like you used our python API, so it would be helpful for us to find a problem if you provide your script.

```
remover = Remover(mode = 'base-nightly',   device = 'cuda:0')
img = Image.open(data_dir + '/' + file).convert('RGB')
img = ImageOps.exif_transpose(img)  
img_no_bg = remover.process(img, threshold = prediction_threshold)
```
Note in the code, the ImageOps.exif_transpose() just rotates the image per the meta data in the jpg file

> Moreover, we also recommend trying base-nightly mode which can be enabled by using --mode base-nightly argument

Yes, I have been using that from the start as I realize this is in active development!

**Comment by eafpres** [comment]
I have confirmed that reducing the image size does affect the proper values for the threshold parameter.  Reducing to 400 x 400, the value of 0.5 is more in the middle of the effects,.

**Comment by plemeri** [comment]
That is true. Actually, we trained our model with the fixed size of 1024 x 1024, so it would produce a more stable result if the given image is hard enough to generate a saliency mask such as your example. Salient object detection dataset are mostly center biased, which means, objects are mostly located on the center of the image, and mostly not occluded nor truncated by the image frame. Otherwise, the network would struggle finding the proper region of saliency. Moreover, if the given image is way more larger or smaller than usual, then it would struggle more. 

Long story short, I think in your case, you might need to resize the input image into a fixed size like you said 400 x 400. 

Also, if you do not need a high quality result in terms of an accurate prediction around the edges of the object, then you might want you use a `mode= 'fast'` option which automatically resizes the input image into 384 x 384 and it is trained with the same size (384 x384). It also consumes less gpu memory and less computational cost. 

Feel free to ask more question. Thanks.

**Comment by eafpres** [comment]
I've found some interesting cases that seem to largely defeat the algorithm. Are you interested to receive those images for development?

**Comment by plemeri** [comment]
I am interested to see those images, but currently I'm already graduated, so I cannot access to any GPU machine to train for my own project. Thank you for your offer btw.

**Comment by eafpres** [comment]
Here are some samples. I'd be grateful for any thoughts you  have.  In these cases, adjusting threshold makes a change but not enough to get a good result.

![image](https://github.com/plemeri/transparent-background/assets/30508983/45b8f247-b271-4bdd-92c9-076eb1ca28c0)
![image](https://github.com/plemeri/transparent-background/assets/30508983/6b2e07c5-d034-4b53-908c-b30e3b5d3044)
![image](https://github.com/plemeri/transparent-background/assets/30508983/15325581-d47f-4dd8-991c-ebed074174d1)
![image](https://github.com/plemeri/transparent-background/assets/30508983/513f3fcb-3ece-4ed5-8c4e-040dc21e3b1d)
![image](https://github.com/plemeri/transparent-background/assets/30508983/e5f061a4-9a59-47d7-ba11-f7764e1d3e63)
![image](https://github.com/plemeri/transparent-background/assets/30508983/ebde00b6-a79f-4300-ab48-b39517bb4562)
![image](https://github.com/plemeri/transparent-background/assets/30508983/4fc627db-a478-4f7f-9d7a-bdf4b76b90fc)
![image](https://github.com/plemeri/transparent-background/assets/30508983/e5a7e858-1d7c-48c0-a418-e588a5759b80)
![image](https://github.com/plemeri/transparent-background/assets/30508983/c1a041a7-232d-423f-abc2-9add16078acf)
![image](https://github.com/plemeri/transparent-background/assets/30508983/17c56b68-93db-4425-b542-1359cc3ddbdb)
![image](https://github.com/plemeri/transparent-background/assets/30508983/9ebbbc43-b95c-4bb9-86bf-7223c233027c)
![image](https://github.com/plemeri/transparent-background/assets/30508983/548e6073-49d4-4981-a28c-955b1ca98791)
![image](https://github.com/plemeri/transparent-background/assets/30508983/a088b30f-dfe4-426c-af4f-e52591de4005)
![image](https://github.com/plemeri/transparent-background/assets/30508983/7029213c-d388-4c01-a59b-a6498fbfe9bd)
